### PR TITLE
Add an 'exec' example. Clean up some websockets code.

### PR DESCRIPTION
**Pull request by brendandburns**
@tintoy 

Thanks!

### CODE REVIEW

File: examples/exec/Exec.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Threading.Tasks;
+using k8s;
+using k8s.Models;
+
+namespace exec
+{
+    internal class Exec
+    {
+        private static void Main(string[] args)
```
Comment (tintoy) [inline examples/exec/Exec.cs]: Note sure what version of C# this is built with but latest supports async main if that makes the code cleaner?

File: src/Kubernetes.ConfigInit.cs [hunk 2]
```diff
@@ -59,10 +60,15 @@ public Kubernetes(KubernetesClientConfiguration config, params DelegatingHandler
                     }
 
 #if NET452
-                    ((WebRequestHandler) HttpClientHandler).ServerCertificateValidationCallback =
-                        CertificateValidationCallBack;
+                    ((WebRequestHandler) HttpClientHandler).ServerCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
```
Comment (tintoy) [inline src/Kubernetes.ConfigInit.cs]: Noted for next time :)

File: src/Kubernetes.WebSocket.cs [hunk 3]
```diff
@@ -226,18 +226,20 @@ public Task<WebSocket> WebSocketNamespacedPodAttachAsync(string name, string @na
 
 #if NETCOREAPP2_1
             if (this.CaCert != null)
+            {
                 webSocketBuilder.ExpectServerCertificate(this.CaCert);
-            else
```
Comment (tintoy) [inline src/Kubernetes.WebSocket.cs]: Aren't these mutually exclusive? I suppose throwing an exception would have been clearer when I first wrote this...

File: src/WebSocketBuilder.NetCoreApp2.1.cs [hunk 4]
```diff
@@ -39,34 +39,10 @@ public WebSocketBuilder AddClientCertificate(X509Certificate2 certificate)
...
-                        );
-                    }
-                }
-                catch (Exception chainException)
-                {
-                    Debug.WriteLine(chainException);
-
-                    return false;
-                }
+                return Kubernetes.CertificateValidationCallBack(sender, serverCertificate, certificate, chain, sslPolicyErrors);
```
Comment (tintoy) [inline src/WebSocketBuilder.NetCoreApp2.1.cs]: Oops! Sorry, must have missed that :)

File: src/Kubernetes.WebSocket.cs (hunk 3 above)
Comment (brendandburns) [reply src/Kubernetes.WebSocket.cs]: Maybe? I sort of lean on "the customer is always right" or "the caller is always right"

If they ship us a config with a ca Cert and noVerifyTLS who are we to judge? We should obey their wishes.

File: examples/exec/Exec.cs (hunk 1 above)
Comment (brendandburns) [reply examples/exec/Exec.cs]: Done (had to move to c# 7.1)

File: src/Kubernetes.WebSocket.cs (hunk 3 above)
Comment (tintoy) [reply src/Kubernetes.WebSocket.cs]: Yeah fair enough, that reasoning makes sense to me :)

My personal approach to API design is "forgiving, but also willing to point out when the caller may be making a mistake". If someone specifies both `CaCert` and `SkipTlsVerify`, which one did they really intend to specify? Is it intuitive to them which one we'd select when there's a conflict like this, and do we document this assumption anywhere?

**But** in this case, correct behaviour is dictated by the K8s client config spec which doesn't prevent them specifying both (so I agree with you that there's not much point in stopping them from doing so).

**Comment by tintoy** [comment]
(sorry, review stuff doesn't show up well on mobile :grin:)

**Pull request merged**